dir.props

```diff
@@ -81,6 +82,7 @@
     <EnvironmentVariables Include="BUILDTOOLS_SKIP_CROSSGEN=1" />
     <!-- New env-var names to replace the ones above over time -->
     <EnvironmentVariables Include="DotNetBuildFromSource=true" />
+    <EnvironmentVariables Include="DotNetRestorePackagesPath=$(PackagesDir)" />
```

Comment:
Interesting that the conventions clash here. I see that NuGet's MSBuild code does call this `PackagesPath` in its code. The NuGet tasks don't seem to stick to `Path`/`File`, using `Directory` and even `Folders` too. Annoying that there isn't a solid convention here. I don't think this needs to be addressed in this PR but pointing it out in case it matters for the Repo API.